### PR TITLE
PL: fix up the storage format for legacy survey summaries

### PR DESCRIPTION
**Pull request by breville**
Followup to https://github.com/code-dot-org/code-dot-org/pull/33458.  

The summaries are all now stored in JSON, and correctly, in `Pd::LegacySurveySummary`'s `data` field.

Here are samples for each type of summary, after parsing the JSON back into ruby hashes:

### csf_intro_post_workshop_all_workshops

```
=> {"how_much_learned_s"=>4.13, "how_motivating_s"=>4.28, "how_clearly_presented_s"=>4.61, "how_interesting_s"=>4.53, "how_often_given_feedback_s"=>4.4, "how_comfortable_asking_questions_s"=>4.65, "how_often_taught_new_things_s"=>4.15, "how_much_participated_s"=>3.97, "how_often_talk_about_ideas_outside_s"=>3.17, "how_often_lost_track_of_time_s"=>3.42, "how_excited_before_s"=>3.45, "overall_how_interested_s"=>4.21, "more_prepared_than_before_s"=>5.36, "know_where_to_go_for_help_s"=>5.45, "suitable_for_my_experience_s"=>5.35, "would_recommend_s"=>5.53, "part_of_community_s"=>5.19, "teacher_engagement"=>3.64, "overall_success"=>5.38, "facilitator_effectiveness"=>4.39, "number_teachers"=>58534, "response_count"=>37833}
```

### csf_intro_post_workshop

```
=> {"how_much_learned_s"=>4.05, "how_motivating_s"=>4.37, "how_clearly_presented_s"=>4.87, "how_interesting_s"=>4.77, "how_often_given_feedback_s"=>4.62, "how_comfortable_asking_questions_s"=>4.84, "how_often_taught_new_things_s"=>4.08, "how_much_participated_s"=>3.91, "how_often_talk_about_ideas_outside_s"=>3.16, "how_often_lost_track_of_time_s"=>3.44, "how_excited_before_s"=>3.48, "overall_how_interested_s"=>4.34, "more_prepared_than_before_s"=>5.62, "know_where_to_go_for_help_s"=>5.73, "suitable_for_my_experience_s"=>5.55, "would_recommend_s"=>5.73, "part_of_community_s"=>5.27, "teacher_engagement"=>3.67, "overall_success"=>5.58, "facilitator_effectiveness"=>4.51, "number_teachers"=>239, "response_count"=>211}
```

### csd_summer_workshops

```
{"How much do you agree or disagree with the following statements about your level of engagement in the workshop? -> I found the activities we did in this workshop interesting and engaging."=>6.16, "How much do you agree or disagree with the following statements about your level of engagement in the workshop? -> I was highly active and participated a lot in the workshop activities."=>6.07, "How much do you agree or disagree with the following statements about your level of engagement in the workshop? -> When I'm not in Code.org workshops, I frequently talk about ideas or content from the workshop with others."=>5.31, "How much do you agree or disagree with the following statements about your level of engagement in the workshop? -> I am definitely planning to make use of ideas and content covered in this workshop in my classroom."=>6.46, "How much do you agree or disagree with the following statements about the workshop overall? -> I feel more prepared to teach the material covered in this workshop than before I came."=>6.35, "How much do you agree or disagree with the following statements about the workshop overall? -> I know where to go if I need help preparing to teach this material."=>6.47, "How much do you agree or disagree with the following statements about the workshop overall? -> This professional development was suitable for my level of experience with teaching CS."=>6.24, "How much do you agree or disagree with the following statements about the workshop overall? -> I feel connected to a community of computer science teachers."=>6.25, "How much do you agree or disagree with the following statements about the workshop overall? -> I would recommend this professional development to others."=>6.51, "How much do you agree or disagree with the following statements about the workshop overall? -> This was the the absolute best professional development I have ever participated in."=>5.67, "During my workshop, {facilitatorName} did the following: -> Demonstrated knowledge of the curriculum."=>6.8, "During my workshop, {facilitatorName} did the following: -> Built and sustained an equitable learning environment in our workshop."=>6.72, "During my workshop, {facilitatorName} did the following: -> Kept the workshop and participants on track."=>6.73, "During my workshop, {facilitatorName} did the following: -> Supported productive workshop discussions."=>6.71, "During my workshop, {facilitatorName} did the following: -> Helped me see ways to create and support an equitable learning environment for my students."=>6.67, "During my workshop, {facilitatorName} did the following: -> Demonstrated a healthy working relationship with their co-facilitator (if applicable)."=>6.8}
```

### csp_summer_workshops

```
{"How much do you agree or disagree with the following statements about your level of engagement in the workshop? -> I found the activities we did in this workshop interesting and engaging."=>6.48, "How much do you agree or disagree with the following statements about your level of engagement in the workshop? -> I was highly active and participated a lot in the workshop activities."=>6.37, "How much do you agree or disagree with the following statements about your level of engagement in the workshop? -> When I'm not in Code.org workshops, I frequently talk about ideas or content from the workshop with others."=>5.89, "How much do you agree or disagree with the following statements about your level of engagement in the workshop? -> I am definitely planning to make use of ideas and content covered in this workshop in my classroom."=>6.59, "How much do you agree or disagree with the following statements about the workshop overall? -> I feel more prepared to teach the material covered in this workshop than before I came."=>6.59, "How much do you agree or disagree with the following statements about the workshop overall? -> I know where to go if I need help preparing to teach this material."=>6.58, "How much do you agree or disagree with the following statements about the workshop overall? -> This professional development was suitable for my level of experience with teaching CS."=>6.67, "How much do you agree or disagree with the following statements about the workshop overall? -> I feel connected to a community of computer science teachers."=>6.63, "How much do you agree or disagree with the following statements about the workshop overall? -> I would recommend this professional development to others."=>6.81, "How much do you agree or disagree with the following statements about the workshop overall? -> This was the the absolute best professional development I have ever participated in."=>6.24, "During my workshop, {facilitatorName} did the following: -> Demonstrated knowledge of the curriculum."=>6.77, "During my workshop, {facilitatorName} did the following: -> Built and sustained an equitable learning environment in our workshop."=>6.81, "During my workshop, {facilitatorName} did the following: -> Kept the workshop and participants on track."=>6.72, "During my workshop, {facilitatorName} did the following: -> Supported productive workshop discussions."=>6.72, "During my workshop, {facilitatorName} did the following: -> Helped me see ways to create and support an equitable learning environment for my students."=>6.73, "During my workshop, {facilitatorName} did the following: -> Demonstrated a healthy working relationship with their co-facilitator (if applicable)."=>6.77}
```

### CODE REVIEW

File: bin/oneoff/generate_legacy_survey_summaries.rb [hunk 1]
```diff
@@ -131,15 +131,17 @@ def snapshot_summer_workshops_from_jotform(course)
...
-
-      key = "facilitator_#{id}_all_ws"
-      rollup[:rollups][key][:averages].each do |string_key, average|
-        # We have found a string_key and an average, now find out the actual string for the key
-        question_text = rollup[:questions][string_key]
-        unless question_text.nil?
-          question_text.gsub!("{facilitatorName}", f.name)
-          facilitator_report[question_text] = average
+      # Do workshop rollups, then facilitator rollups.
+      [false, true].each do |facilitator_rollups|
```
Comment (molly-moen) [inline bin/oneoff/generate_legacy_survey_summaries.rb]: what does [false, true] do here?

Comment (molly-moen) [inline bin/oneoff/generate_legacy_survey_summaries.rb]: nit: can you make the facilitator_rollups variable name more clear? Ex only_facilitator_questions?

**Review by molly-moen — APPROVED** [review]

File: bin/oneoff/generate_legacy_survey_summaries.rb (hunk 1 above)
Comment (breville) [reply bin/oneoff/generate_legacy_survey_summaries.rb]: Sure thing.  

I did struggle over this particular issue.  The comment above is perhaps just as useful: it borrows the facilitator- and workhop-rollups naming from these lines: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/lib/pd/survey_pipeline/survey_pipeline_helper.rb#L36-L37

It's one challenge of using a boolean parameter to actually switch between different modes.

Comment (molly-moen) [reply bin/oneoff/generate_legacy_survey_summaries.rb]: thanks! the comment was definitely helpful, without that I would have been very confused :)

**Comment by bencodeorg** [comment]
What happened before you added the `.to_json` when you tried to parse back into Ruby? Been unsure of whether to create records that store a ruby hash in a JSON column using `.to_json` or not on the contact rollups side as well.

**Comment by bencodeorg** [comment]
Also, lol at a hash key that includes `->` 😆

**Comment by breville** [comment]
@bencodeorg IIRC, it was storing them as the equivalent of a ruby hash converted to a string using `.to_s`.

**Pull request merged**